python-package/lightgbm/engine.py

```diff
@@ -109,6 +109,8 @@ def train(params, train_set, num_boost_round=100,
     """
     # create predictor first
     params = copy.deepcopy(params)
+    if fobj is not None:
+        params['objective'] = 'none'
```

Comment:
This fix is similar to the following one:
https://github.com/Microsoft/LightGBM/blob/843c83822508d5e89b63659ad582aedf8a304ed7/python-package/lightgbm/basic.py#L1789-L1790
and it allows to execute the following snippet:
```
from sklearn.datasets import load_digits
import lightgbm as lgb
import numpy as np

def custom_obj(preds, train_data):
    return np.zeros(preds.shape), np.zeros(preds.shape)

X, y = load_digits(3, True)
lgb_data = lgb.Dataset(X, y)
lgb.train({'num_class': 3}, lgb_data, fobj=custom_obj)

---------------------------------------------------------------------------
LightGBMError                             Traceback (most recent call last)
<ipython-input-3-25adeff81fd1> in <module>
      1 X, y = load_digits(3, True)
      2 lgb_data = lgb.Dataset(X, y)
----> 3 lgb.train({'num_class': 3}, lgb_data, fobj=custom_obj)

C:\Program Files\Anaconda3\lib\site-packages\lightgbm\engine.py in train(params, train_set, num_boost_round, valid_sets, valid_names, fobj, feval, init_model, feature_name, categorical_feature, early_stopping_rounds, evals_result, verbose_eval, learning_rates, keep_training_booster, callbacks)
    193     # construct booster
    194     try:
--> 195         booster = Booster(params=params, train_set=train_set)
    196         if is_valid_contain_train:
    197             booster.set_train_data_name(train_data_name)

C:\Program Files\Anaconda3\lib\site-packages\lightgbm\basic.py in __init__(self, params, train_set, model_file, silent)
   1533             self.handle = ctypes.c_void_p()
   1534             _safe_call(_LIB.LGBM_BoosterCreate(
-> 1535                 train_set.construct().handle,
   1536                 c_str(params_str),
   1537                 ctypes.byref(self.handle)))

C:\Program Files\Anaconda3\lib\site-packages\lightgbm\basic.py in construct(self)
    988                                 init_score=self.init_score, predictor=self._predictor,
    989                                 silent=self.silent, feature_name=self.feature_name,
--> 990                                 categorical_feature=self.categorical_feature, params=self.params)
    991             if self.free_raw_data:
    992                 self.data = None

C:\Program Files\Anaconda3\lib\site-packages\lightgbm\basic.py in _lazy_init(self, data, label, reference, weight, group, init_score, predictor, silent, feature_name, categorical_feature, params)
    780             self.__init_from_csc(data, params_str, ref_dataset)
    781         elif isinstance(data, np.ndarray):
--> 782             self.__init_from_np2d(data, params_str, ref_dataset)
    783         elif isinstance(data, list) and len(data) > 0 and all(isinstance(x, np.ndarray) for x in data):
    784             self.__init_from_list_np2d(data, params_str, ref_dataset)

C:\Program Files\Anaconda3\lib\site-packages\lightgbm\basic.py in __init_from_np2d(self, mat, params_str, ref_dataset)
    842             c_str(params_str),
    843             ref_dataset,
--> 844             ctypes.byref(self.handle)))
    845         return self
    846 

C:\Program Files\Anaconda3\lib\site-packages\lightgbm\basic.py in _safe_call(ret)
     44     """
     45     if ret != 0:
---> 46         raise LightGBMError(decode_string(_LIB.LGBM_GetLastError()))
     47 
     48 

LightGBMError: Number of classes must be 1 for non-multiclass training
```

This is due to that multiclass check is performed during Booster initialization.